src/utils/sformat.fs

```diff
@@ -54,65 +54,73 @@ namespace Microsoft.FSharp.Text.StructuredFormat
 
     [<NoEquality; NoComparison>]
 #if COMPILER
+    type internal BoxRange = BoxRange of obj
```

Comment:
@saul I'm ok with moving the file unless it proves very difficult, it's totally fair to push back on this kind of hack.  

This file is a bit odd because it also forms part of FSHarp.Core - we should probably just duplicate it out so that the compiler and FSHarp.Core are independent code bases, and we don't have all the whacky ``#if COMPILER``